orcid-web/src/main/java/org/orcid/frontend/web/controllers/PublicRecordController.java

```diff
@@ -258,6 +315,203 @@ PublicRecord getRecord(String orcid) {
         return publicRecord;
     }
 
+    @RequestMapping(value = "/{orcid:(?:\\d{4}-){3,}\\d{3}[\\dX]}/summary.json", method = RequestMethod.GET)
+    public @ResponseBody
+    RecordSummary getSummaryRecord(@PathVariable("orcid") String orcid) {
+        RecordSummary recordSummary = new RecordSummary();
+        Boolean isDeprecated = false;
+
+        try {
+            // Check if the profile is deprecated or locked
+            orcidSecurityManager.checkProfile(orcid);
+        } catch (LockedException | DeactivatedException e) {
+            recordSummary.setName(localeManager.resolveMessage("public_profile.deactivated.given_names") + " "
+                    + localeManager.resolveMessage("public_profile.deactivated.family_name"));
+            return recordSummary;
+        } catch (OrcidNotClaimedException e) {
+            recordSummary.setName(localeManager.resolveMessage("orcid.reserved_for_claim"));
+            return recordSummary;
+        } catch (OrcidDeprecatedException e) {
+            isDeprecated = true;
+        } catch (OrcidNoResultException e) {
+            return recordSummary;
+        }
+
+        recordSummary = getSummary(orcid);
+        if (isDeprecated) {
```

Comment:
Hey @DanielPalafox, lets make this check before the `getSummary(orcid)` so we dont waste resources fetching the info if the record is deprecated

Reply:
And add an else clause where you call the getSummary in case it is not deprecated

Reply:
Done!